src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfComplexTypesTests.cs

```diff
@@ -158,25 +158,58 @@ public void TestUdfWithRowType()
         [Fact]
         public void TestUdfWithReturnAsRowType()
         {
-            // UDF with return as RowType throws a following exception:
-            // Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. 
-            // --->System.ArgumentException: System.Object is not supported.
-            // at Microsoft.Spark.Utils.UdfUtils.GetReturnType(Type type) in Microsoft.Spark\Utils\UdfUtils.cs:line 142
-            // at Microsoft.Spark.Utils.UdfUtils.GetReturnType(Type type) in Microsoft.Spark\Utils\UdfUtils.cs:line 136
-            // at Microsoft.Spark.Sql.Functions.CreateUdf[TResult](String name, Delegate execute, PythonEvalType evalType) in Microsoft.Spark\Sql\Functions.cs:line 4053
-            // at Microsoft.Spark.Sql.Functions.CreateUdf[TResult](String name, Delegate execute) in Microsoft.Spark\Sql\Functions.cs:line 4040
-            // at Microsoft.Spark.Sql.Functions.Udf[T, TResult](Func`2 udf) in Microsoft.Spark\Sql\Functions.cs:line 3607
-            Assert.Throws<ArgumentException>(() => Udf<string, Row>(
-                (str) =>
-                {
-                    var structFields = new List<StructField>()
+            // Single GenericRow
+            var schema1 = new StructType(new[]
+            {
+                new StructField("col1", new IntegerType()),
+                new StructField("col2", new StringType())
+            });
+            Func<Column, Column> udf1 = Udf<string>(
+                str => new GenericRow(new object[] { 1, "abc" }), schema1);
+
+            Row[] rows1 = _df.Select(udf1(_df["name"])).Collect().ToArray();
+            Assert.Equal(3, rows1.Length);
+
+            foreach (Row row in rows1)
+            {
+                Assert.Equal(2, row.Size());
+                Assert.Equal(1, row.GetAs<int>("col1"));
+                Assert.Equal("abc", row.GetAs<string>("col2"));
+            }

```

Comment:
Let's put these in {}'s and rename vars like `schema1`, `rows1`, etc to `schema`, `rows`, etc.

Reply:
Sure, added!